pkg/boot/linux/load_linux_arm64.go

```diff
@@ -0,0 +1,235 @@
+		}
+	}
+
+	kImage, err := image.ParseFromBytes(kernelBuf)
+	if err != nil {
+		return fmt.Errorf("parse arm64 Image from bytes: %v", err)
+	}
+
+	if kImage.Header.TextOffset > 0 {
+		// TODO(10000TB): load kernel by text_offset.
+		return fmt.Errorf("non zero text offset is not supported yet.")
+	} else {
+		// TextOffset as 0.
+		if kernelRange, err = kmem.AddKexecSegmentExplicit(b, ah.ImageSize); err != nil {
+			return fmt.Errorf("add kernel segment: %v", err)
+		}
+	}
+	Debug("Added %d byte (size %d) kernel at %s", len(kernelBuf), ah.ImageSize, kernelRange)
+
+	var ramfsBuf []byte
+	if ramfs != nil {
+		if opts.MmapRamfs {
+			Debug("Mmap ramfs file to virtual buffer...")
+			ramfsBuf, ummap, err := mmap(ramfs)
+			if err != nil {
+				return fmt.Errorf("mmap ramfs: %v", err)
+			}
+			defer func() {
+				if err = ummap(); err != nil {
+					return Debug("Ummap ramfs failed: %v", err)
+				}
+			}()
+		} else {
+			Debug("Read ramfs from file...")
+			ramfsBuf, err = uio.ReadAll(ramfs)
+			if err != nil {
+				return fmt.Errorf("read ramfs from file: %v", err)
+			}
+		}
+	}
+
+	// NOTE(10000TB): This need be placed after kernel by convention.
+	if ramfsRange, err = kmem.AddKexecSegment(rb); err != nil {
+		return fmt.Errorf("add initramfs segment: %v", err)
+	}
+	Debug("Added %d byte initramfs at %s", len(ramfsBuf), ramfsRange)
+
+	ramfsStart := make([]byte, 8)
+	binary.BigEndian.PutUint64(ramfsAddr, uint64(ramfsRange.Start))
+	chosen.UpdateProperty("linux,initrd-start", ramfsStart)
+	ramfsEnd = make([]byte, 8)
+	binary.BigEndian.PutUint64(ramfsEnd, uint64(ramfsRange.Start)+uint64(ramfsRange.Size))
+	chosen.UpdateProperty("linux,initrd-end", ramfsEnd)
+
+	Debug("Kernel cmdline to append: %s", cmdline)
+	if len(cmdline) > 0 {
+		cmdlineBuf := append([]byte(cmdline), byte(0))
+		chosen.UpdateProperty("bootargs", cmdlineBuf)
+	} else {
+		chosen.RemoveProperty("bootargs")
+	}
+
+	dtbBuffer := &bytes.Buffer{}
+	_, err = fdt.Write(dtbBuffer)
+	if err != nil {
+		return fmt.Errorf("flattening device tree: %v", err)
+	}
+	dtbBuf := dtbBuffer.Bytes()
+	if dtbRange, err = kmem.AddKexecSegment(dtbBuf); err != nil {
+		return fmt.Errorf("add device tree segment: %w", err)
+	}
+	Debug("Added %d byte device tree at %s", len(dtbBuf), dtbRange)
+
+	// Trampoline.
+	//
+	// We need a trampoline to pass the DTB to the kernel; because
+	// we'll use this code as our entry point, it also needs to know
+	// the real entry point to kernel.
+	//
+	// TODO(10000TB): this assumes a little endian kernel, support
+	// big endian if needed per flag.
+	kernelEntry := kernelRange.Start
+	dtbBase := dtbRange.Start
+	var trampoline [10]uint32
+	// Instruction encoding per
+	// "Arm Architecture Reference Manual Armv8, for Armv8-A architecture
+	// profile" [ ARM DDI 0487E.a (ID070919) ]
+	//
+	// LDR(literal), load kernel entry from 6 * 4 bytes away up from this
+	// instruction, which is from trampoline[6].
+	trampoline[0] = 0x580000c4 // ldr x4, arm64_kernel_entry
```

Comment:
Might I suggest
  ldr x4, #0x18  (PC relative: trampoline[6 and 7])
  ldr x0, #0x20  (PC relative: trampoline[8 and 9])
as the instruction comments?

Reply:
Done. (replaced the inline comment w/ suggestions, but kept the comment above)

That is less confusing than using the symbol / label, which we don't set anywhere. `arm64_kernel_entry ` is identical to that from kexec tools, which may lead to confusion we are passing it via symbols

Reply:
`#0x20` need be replaced w/  `#0x1c` actually, as there is 7 uint32, 28 bytes away.


 111    0  0000 --- 7 is the imm

Reply:
"#0x20" doesn't mean "add 32" in this context, it's loading from address 0x20 (assuming trampoline[0] starts at 0).

I'm not 100% sure this is the correct syntax for PC-relative in Aarch64, but it's the decode I get from hell-storm.org/online/Online-Assembler-and-Disassembler